pkg/webhook/v1beta1/experiment/validator/validator.go

```diff
@@ -420,79 +472,93 @@ func validatePatchJob(runSpec *unstructured.Unstructured, job interface{}, jobTy
 	return nil
 }
 
-func (g *DefaultValidator) validateMetricsCollector(inst *experimentsv1beta1.Experiment) error {
+func (g *DefaultValidator) validateMetricsCollector(inst *experimentsv1beta1.Experiment) field.ErrorList {
+	var allErrs field.ErrorList
 	mcSpec := inst.Spec.MetricsCollectorSpec
 	mcKind := mcSpec.Collector.Kind
 	for _, mc := range mccommon.AutoInjectMetricsCollectorList {
 		if mcKind != mc {
 			continue
 		}
 		if _, err := g.GetMetricsCollectorConfigData(mcKind); err != nil {
-			return fmt.Errorf("GetMetricsCollectorConfigData failed: %v", err)
+			allErrs = append(allErrs, field.Invalid(metricsCollectorPath.Child("collector").Child("kind"),
+				mcKind, fmt.Sprintf("GetMetricsCollectorConfigData failed: %v", err)))
 		}
 		break
 	}
 	// TODO(hougangliu): log warning message if some field will not be used for the metricsCollector kind
 	switch mcKind {
 	case commonapiv1beta1.NoneCollector, commonapiv1beta1.StdOutCollector:
-		return nil
+		return allErrs
 	case commonapiv1beta1.FileCollector:
 		if mcSpec.Source == nil || mcSpec.Source.FileSystemPath == nil ||
 			mcSpec.Source.FileSystemPath.Kind != commonapiv1beta1.FileKind || !filepath.IsAbs(mcSpec.Source.FileSystemPath.Path) {
-			return fmt.Errorf("file path where metrics file exists is required by .spec.metricsCollectorSpec.source.fileSystemPath.path")
+			allErrs = append(allErrs, field.Required(metricsSourcePath.Child("fileSystemPath").Child("path"),
+				"file path where metrics file exists is required by .spec.metricsCollectorSpec.source.fileSystemPath.path"))
 		}
 		// Format
 		fileFormat := mcSpec.Source.FileSystemPath.Format
 		if fileFormat != commonapiv1beta1.TextFormat && fileFormat != commonapiv1beta1.JsonFormat {
-			return fmt.Errorf("format of metrics file is required by .spec.metricsCollectorSpec.source.fileSystemPath.format")
+			allErrs = append(allErrs, field.Required(metricsSourcePath.Child("fileSystemPath").Child("format"),
+				"format of metrics file is required by .spec.metricsCollectorSpec.source.fileSystemPath.format"))
```

Comment:
@ckcd How the error message will look like that we send to the users ?
Do we need to simplify it as follows:
`format of metrics file is required for metrics collector` ?

Reply:
@andreyvelich Yes the error message looks a little redundant:
```
[spec.metricsCollectorSpec.source.fileSystemPath.format: Required value: format of metrics file is required by .spec.metricsCollectorSpec.source.fileSystemPath.format]
```

I have made changes to remove redundant information according to your suggestion. Thanks.